src/reducers/dialogVisibility.js

```diff
@@ -0,0 +1,25 @@
+import Immutable from 'immutable'
+import { actionReducer } from './utils'
+
+const dialogVisibility = actionReducer(Immutable.fromJS({ showVmDialog: false, showVmDetail: false, showEditTemplate: false }), {
```

Comment:
Somewhere else, I'm using this structure:
{
topLevel: 'type as String constant  or Object with details'
// if really required later, as a hack we can add another key along the topLevel, it would be still manageble
type: {
 // type specific map of details
 // if complex, then tree structure as the topLevel
 // but oVirt web admin is a good example that there's no need for very complex 2nd or 3rd level 
}
}

The reudcer (read: at one place) handles potential complexities of dependencies.

It works well. The ImmutableJS is a big benefit.